### PR TITLE
created Rotator and ManualRotator, set RobotMap ports

### PR DESCRIPTION
**Pull request by fielia**

### CODE REVIEW

File: src/main/java/org/recordrobotics/munchkin/commands/manual/ManualRotator.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package org.recordrobotics.munchkin.commands.manual;
+
+import org.recordrobotics.munchkin.control.IControlInput;
+import org.recordrobotics.munchkin.subsystems.Rotator;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class ManualRotator extends CommandBase {
+	Rotator _rotator;
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/commands/manual/ManualRotator.java:9]: Missing access modifiers

File: src/main/java/org/recordrobotics/munchkin/commands/manual/ManualRotator.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	public ManualRotator(Rotator rotator, IControlInput control) {
+		_rotator = rotator;
+		_controls = control;
+	}
+	
+	@Override
+	public void execute() {
+		double speed = _controls.getRotate();
+		if (speed < 0.15) {
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/commands/manual/ManualRotator.java:20]: This won't work for negatives!!
Also magic number

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import org.recordrobotics.munchkin.RobotMap;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Rotator extends SubsystemBase {
+	private CANSparkMax leftMotor = new CANSparkMax(RobotMap.Rotator.LEFT_MOTOR_PORT, MotorType.kBrushless);
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:13]: Doesn't follow naming convention, see PMD error

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    private CANSparkMax rightMotor = new CANSparkMax(RobotMap.Rotator.RIGHT_MOTOR_PORT, MotorType.kBrushless);
+    private RelativeEncoder leftEncoder = leftMotor.getEncoder();
+    private RelativeEncoder rightEncoder = rightMotor.getEncoder();
+	private DigitalInput forwardLimit = new DigitalInput(RobotMap.Rotator.FWD_LIMIT_PORT);
+    private DigitalInput backwardLimit = new DigitalInput(RobotMap.Rotator.BWD_LIMIT_PORT);
+
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:19]: Subsystem constructor should stop the motor

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		leftMotor.set(0);
+		rightMotor.set(0);
+	}
+
+	/**
+	 * gets position from encoders
+	 * @return position
+	 */
+	public double getPosition() {
+		return (leftEncoder.getPosition() - rightEncoder.getPosition()) / 2;
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:33]: Add a comment explaining this calculation (why there is a minus sign)

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+	/**
+     * rotate lift forwards and backwards
+     * @param speed speed
+     */
+    public void rotate(double speed) {
+        if ((speed > 0 && forwardLimit.get()) || (speed < 0 && backwardLimit.get())) {
+            leftMotor.set(Subsystems.limitSpeed(speed));
+            rightMotor.set(-Subsystems.limitSpeed(speed));
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:51]: It makes more sense to pass -speed to the function.

**Review by vladaviedov — CHANGES_REQUESTED** [review]

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+public class Rotator extends SubsystemBase {
+	private CANSparkMax _leftMotor = new CANSparkMax(RobotMap.Rotator.LEFT_MOTOR_PORT, MotorType.kBrushless);
+	private CANSparkMax _rightMotor = new CANSparkMax(RobotMap.Rotator.RIGHT_MOTOR_PORT, MotorType.kBrushless);
+	private RelativeEncoder _leftEncoder = _leftMotor.getEncoder();
+	private RelativeEncoder _rightEncoder = _rightMotor.getEncoder();
+	private DigitalInput _forwardLimit = new DigitalInput(RobotMap.Rotator.FWD_LIMIT_PORT);
+	private DigitalInput _backwardLimit = new DigitalInput(RobotMap.Rotator.BWD_LIMIT_PORT);
+
+	public Rotator() {
+		_leftMotor.set(0);
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:21]: You have a function for this

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+		_rightMotor.set(0);
+	}
+
+	/**
+	 * gets position from encoders
+	 * @return position
+	 */
+	public double getPosition() {
+		// right encoder will be incrementing opposite to left encoder. because left motor spins in the same direction as the input, right encoder is subtracted.
```
Comment (vladaviedov) [inline src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:38]: ?

**Review by vladaviedov — COMMENTED** [review]
That comment is very confusing

File: src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java (hunk 7 above)
Comment (fielia) [reply src/main/java/org/recordrobotics/munchkin/subsystems/Rotator.java:21]: the function throws a pmd error, and idk how to fix it. i just deleted stop()

**Review by vladaviedov — COMMENTED** [review]
Looks good except the comment about encoder values

**Review by vladaviedov — APPROVED** [review]

**Pull request merged**